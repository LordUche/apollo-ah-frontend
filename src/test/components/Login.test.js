/* eslint-disable react/jsx-filename-extension */
import '@babel/polyfill';
import React from 'react';
import setup from '../setup/index';
import  ConnectedLogin from '../../components/Login';

// eslint-disable-next-line no-undef
const mockFn = jest.fn();

location = {
  pathname: '',
};

describe('<UserForm />', () => {

  it('should respond to input change and alter state (email) of component', () => {
    const wrapper = setup(<ConnectedLogin location={location} />);
    const loginWrapper = wrapper.find('Login');
    wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'wrestle@wwe.com' } });
    wrapper.update();
    expect(loginWrapper.state('email')).toBe('wrestle@wwe.com');
    expect(wrapper.find('#email').props().value).toBe('wrestle@wwe.com');
  });
  it('should respond to input change and alter state (password) of component', () => {
    const wrapper = setup(<ConnectedLogin location={location} />);
    const loginWrapper = wrapper.find('Login');
    wrapper.find('#password').simulate('change', { target: { name: 'password', value: '12345' } });
    wrapper.update();
    expect(loginWrapper.state('password')).toBe('12345');
    expect(wrapper.find('#password').props().value).toBe('12345');
  });
  it('should submit form', () => {
    const wrapper = setup(<ConnectedLogin location={location} />);
    const loginWrapper = wrapper.find('Login');
    loginWrapper.find('.login__input__section form').simulate('submit', { preventDefault: mockFn });
    expect(mockFn).toHaveBeenCalled();
  });
  it('should submit form', () => {
    const wrapper = setup(<ConnectedLogin location={location} />);
    wrapper.find('#myModal form').simulate('submit', { preventDefault: mockFn });
    expect(mockFn).toHaveBeenCalled();
  });
});
