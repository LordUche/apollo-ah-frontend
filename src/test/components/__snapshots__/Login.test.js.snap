// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> Should render all elements 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Login)
        location={
          Object {
            "pathname": "",
          }
        }
      >
        <Login
          isLoading={false}
          location={
            Object {
              "pathname": "",
            }
          }
          loginStatus="failure"
          message=""
          processLogin={[Function]}
        >
          <UserForm
            email=""
            handleChange={[Function]}
            handleLogin={[Function]}
            isLoading={false}
            location=""
            message=""
            password=""
            resetState={[Function]}
            status=""
            usernameInput={<div />}
          >
            <div
              className="background min-vh-100"
            >
              <Connect(ResetPassword)>
                <ResetPassword
                  isLoading={false}
                  message=""
                  modalAction={[Function]}
                  passwordResetRequest={[Function]}
                >
                  <Modal
                    clearMessage={[Function]}
                    title="Recover Password!"
                  >
                    <div
                      className="modal"
                      id="myModal"
                      onClick={[Function]}
                    >
                      <div
                        className="modal-dialog"
                      >
                        <div
                          className="modal_content"
                          onClick={[Function]}
                        >
                          <div
                            className="cancelTop"
                          >
                            <button
                              className="close"
                              data-dismiss="modal"
                              onClick={[Function]}
                              type="button"
                            >
                              ×
                            </button>
                          </div>
                          <div
                            className="modal_body"
                          >
                            <div
                              className="modal_header"
                            >
                              <h4
                                className="modal_title"
                              >
                                Recover Password!
                              </h4>
                            </div>
                            <p
                              className="reset_password_paragraph"
                            >
                              Enter the email address you used when you created your account. You will receive instructions to reset your password in your mail box.
                            </p>
                            <form
                              className="reset-password-group "
                              onSubmit={[Function]}
                            >
                              <div
                                className="form-group"
                              >
                                <input
                                  autoComplete="true"
                                  className="reset_form_control btn-block"
                                  name="email"
                                  onChange={[Function]}
                                  placeholder="Enter email"
                                  required={true}
                                  type="email"
                                />
                                <span
                                  className="input-danger mt-2"
                                >
                                  A valid email is required
                                </span>
                                <button
                                  className="reset_btn btn-block mt-3"
                                  type="submit"
                                >
                                  Request reset link
                                </button>
                              </div>
                            </form>
                            <div />
                          </div>
                        </div>
                      </div>
                    </div>
                  </Modal>
                </ResetPassword>
              </Connect(ResetPassword)>
              <div
                className="container-fluid"
              >
                <div
                  className="row"
                >
                  <div
                    className="login"
                  >
                    <div
                      className="col-xs-12 col-sm-6 col-md-6 col-lg-6"
                    >
                      <div
                        className="login__nav"
                      >
                        <Link
                          href="/"
                          replace={false}
                          to="/"
                        >
                          <a
                            href="/"
                            onClick={[Function]}
                          >
                            ←
                          </a>
                        </Link>
                      </div>
                      <div
                        className="login__content"
                      >
                        <div
                          className="login__intro"
                        >
                          <h3
                            className="text-center"
                          >
                            Signup
                          </h3>
                          <p>
                            Please fill in your email and password to gain access to your account.
                          </p>
                        </div>
                        <div
                          className="login__input__section"
                        >
                          <form
                            onSubmit={[Function]}
                          >
                            <div
                              className="form-group"
                            >
                              <div
                                className="login__input__field"
                              >
                                <label
                                  htmlFor="email"
                                >
                                  Email
                                </label>
                                <input
                                  autoComplete="true"
                                  className="form-control"
                                  id="email"
                                  name="email"
                                  onChange={[Function]}
                                  type="email"
                                  value=""
                                />
                              </div>
                              <div />
                              <div
                                className="login__input__field"
                              >
                                <div
                                  className="login__forgot-password"
                                >
                                  <label
                                    htmlFor="password"
                                  >
                                    Password
                                  </label>
                                  <button
                                    className="btn btn-link login__reset__btn "
                                    data-target="#myModal"
                                    data-toggle="modal"
                                    type="button"
                                  >
                                    Forgot password?
                                  </button>
                                </div>
                                <input
                                  autoComplete="true"
                                  className="form-control"
                                  id="password"
                                  name="password"
                                  onChange={[Function]}
                                  type="password"
                                  value=""
                                />
                              </div>
                              <div>
                                <button
                                  className="form-control login__btn--large"
                                  type="submit"
                                >
                                  Join Authors Haven
                                </button>
                              </div>
                            </div>
                          </form>
                        </div>
                        <div
                          className="login__socials"
                        >
                          <div
                            className="row login__socials__intro"
                          >
                            <hr />
                            <p>
                              Or login using
                            </p>
                          </div>
                          <Connect(SocialLogin)>
                            <SocialLogin
                              handleLogin={[Function]}
                              isAuthenticated={false}
                            >
                              <div
                                className="row login__socials__content"
                              >
                                <div
                                  className="login__socials__icons"
                                >
                                  <SocialButton
                                    altText=""
                                    icon={Object {}}
                                    onClick={[Function]}
                                  >
                                    <button
                                      className="btn"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      <img
                                        alt=""
                                        src={Object {}}
                                      />
                                    </button>
                                  </SocialButton>
                                  <SocialButton
                                    altText=""
                                    icon={Object {}}
                                    onClick={[Function]}
                                  >
                                    <button
                                      className="btn"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      <img
                                        alt=""
                                        src={Object {}}
                                      />
                                    </button>
                                  </SocialButton>
                                </div>
                              </div>
                            </SocialLogin>
                          </Connect(SocialLogin)>
                        </div>
                      </div>
                    </div>
                    <div
                      className="d-none d-sm-flex col-sm-6 col-md-6 col-lg-6 banner"
                    >
                      <div
                        className="login__banner"
                      >
                        <div
                          className="login__banner__content"
                        >
                          <hr />
                          <div>
                            <h3>
                              Hello,
                            </h3>
                            <p>
                              Log in to your account to start reading and writing beautiful articles.
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </UserForm>
        </Login>
      </Connect(Login)>
    </Router>
  </BrowserRouter>
</Provider>
`;
